src/main/kotlin/io.javalin/HaltException.kt

```diff
@@ -0,0 +1,30 @@
+/*
+ * Javalin - https://javalin.io
+ * Copyright 2017 David Åse
+ * Licensed under Apache 2.0: https://github.com/tipsy/javalin/blob/master/LICENSE
+ */
+
+package io.javalin
+
+import javax.servlet.http.HttpServletResponse
+
+class HaltException : RuntimeException {
+    var statusCode = HttpServletResponse.SC_OK
+    var body = "Execution halted"
+
+    constructor() {}
+
+    constructor(statusCode: Int) {
```

Comment:
you don't need all these constructors, you can use default values

```kotlin
class HaltException(val statusCode = HttpServletResponse.SC_OK, body = "Execution halted")  : RuntimeException
```
and that's your entire class :)